apps/api/src/goods-receipts/dto/create-goods-receipt.dto.ts

```diff
@@ -0,0 +1,28 @@
+import { IsNotEmpty, IsNumber, ValidateNested, ArrayMinSize } from 'class-validator';
+import { Type, Transform } from 'class-transformer';
+
+class GoodsReceiptLineItemDto {
+  id: number;
+  
+  @IsNotEmpty()
+  @IsNumber()
+  item_id: number;
+
+  @IsNotEmpty()
+  @IsNumber()
+  quantity_received: number;
+}
+
+export class CreateGoodsReceiptDto {
+  @IsNotEmpty()
+  @IsNumber()
+  purchase_order_id: number;
+
+  @Transform(({ value }) => new Date(value).toISOString())
+  received_date: string;
+
+  @ValidateNested({ each: true })
+  @ArrayMinSize(1)
+  @Type(() => GoodsReceiptLineItemDto)
+  line_items: GoodsReceiptLineItemDto[];
+}
```

Comment:
nit: lots of missing newlines at end of file. Prettier can help avoid this

Reply:
Will make sure to have prettier format my code on save!